libs/ngx-mime/src/lib/core/alto-service/alto.service.spec.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { cold } from 'jasmine-marbles';
+import { CanvasServiceStub } from '../../test/canvas-service-stub';
+import { IiifManifestServiceStub } from '../../test/iiif-manifest-service-stub';
+import { CanvasService } from '../canvas-service/canvas-service';
+import { IiifManifestService } from '../iiif-manifest-service/iiif-manifest-service';
+import { MimeViewerIntl } from '../intl/viewer-intl';
+import { testAlto } from './../../test/testAltos';
+import { AltoService } from './alto.service';
+
+describe('AltoService', () => {
+  const debounceTime = 200;
+  let service: AltoService;
+  let httpTestingController: HttpTestingController;
+  let iiifManifestService: any;
+  let canvasService: any;
+  let intl: MimeViewerIntl;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        MimeViewerIntl,
+        { provide: CanvasService, useClass: CanvasServiceStub },
+        { provide: IiifManifestService, useClass: IiifManifestServiceStub },
+      ],
+    });
+    service = TestBed.inject(AltoService);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    iiifManifestService = TestBed.inject(IiifManifestService);
+    canvasService = TestBed.inject(CanvasService);
+    intl = TestBed.inject(MimeViewerIntl);
+    setUpSpy();
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+    service.destroy();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should load current alto on canvas change', fakeAsync(() => {
+    service.initialize();
+    iiifManifestService.load('fakeUrl').subscribe(() => {
+      waitForDebounce();
+      mockAltoRequest();
+      expectAltoToBeDefined();
+    });
+  }));
```

Comment:
From test description I would expect a call to a canvas change function in the test, but cannot find one.

Reply:
Fixed. Changing to canvas index 2.

Reply:
I have changed the test description and added another test for canvas change